src/game/common/skirmishbattlehonors.cpp

```diff
@@ -27,3 +39,617 @@ int SkirmishBattleHonors::Get_Endurance_Medal(Utf8String name, int level) const
+    int row = 1;
+    int column = 0;
+
+    const Image *image;
+    if (BattleHonors->Get_China_Campaign_Complete(DIFFICULTY_HARD)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("ChinaCampaign_G");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_CHINA, &row, &column, 0, 0);
+    } else if (BattleHonors->Get_China_Campaign_Complete(DIFFICULTY_NORMAL)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("ChinaCampaign_S");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_CHINA, &row, &column, 0, 0);
+    } else if (BattleHonors->Get_China_Campaign_Complete(DIFFICULTY_EASY)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("ChinaCampaign_B");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_CHINA, &row, &column, 0, 0);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("ChinaCampaign_B");
+        Insert_Battle_Honor(listbox, image, (char)(honors >> 0xC) & 1, CAMPAIGN_CHINA, &row, &column, 0, 0);
+    }
+
+    if (BattleHonors->Get_GLA_Campaign_Complete(DIFFICULTY_HARD)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("GLACampaign_G");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_GLA, &row, &column, 0, 0);
+    } else if (BattleHonors->Get_GLA_Campaign_Complete(DIFFICULTY_NORMAL)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("GLACampaign_S");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_GLA, &row, &column, 0, 0);
+    } else if (BattleHonors->Get_GLA_Campaign_Complete(DIFFICULTY_EASY)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("GLACampaign_B");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_GLA, &row, &column, 0, 0);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("GLACampaign_B");
+        Insert_Battle_Honor(listbox, image, (char)(honors >> 0xD) & 1, CAMPAIGN_GLA, &row, &column, 0, 0);
+    }
+
+    if (BattleHonors->Get_USA_Campaign_Complete(DIFFICULTY_HARD)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("USACampaign_G");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_USA, &row, &column, 0, 0);
+    } else if (BattleHonors->Get_USA_Campaign_Complete(DIFFICULTY_NORMAL)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("USACampaign_S");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_USA, &row, &column, 0, 0);
+    } else if (BattleHonors->Get_USA_Campaign_Complete(DIFFICULTY_EASY)) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("USACampaign_B");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_USA, &row, &column, 0, 0);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("USACampaign_B");
+        Insert_Battle_Honor(listbox, image, (char)(honors >> 0xB) & 1, CAMPAIGN_USA, &row, &column, 0, 0);
+    }
+
+    bool bronze = false;
+    bool silver = false;
+    bool gold = false;
+    for (int i = 0; i < 9; i++) {
```

Comment:
I wonder if we have a constant for this number.

Reply:
I don't see anything related. Each challenge campaign general has either 7 or 8 missions.